### PR TITLE
Improve local development with Docker compose

### LINKED ISSUE #6: docker support?
Can you add a Dockerfile and a docker-compose.yml file to deploy an instance?

### PR DESCRIPTION
**Pull request by tracker1**
Closes #6

- Add Dockerfile for App
- Add docker-compose.yml for App and dependencies
- Update app_config to use environment variables for dependencies
- Add compose instructions to README.md

You can now run this application locally with:

```
docker-compose up
```

Application: http://localhost:8111/
Redis Commander: http://localhost:8112/

### CODE REVIEW

**Comment by tracker1** [comment]
@fallanic btw, didn't mean to change as much to the README, my VS Code is set to auto-format on save, so updated the markdown formatting, `#` for heading prefixes, `-` for lists, etc.

**Comment by tracker1** [comment]
@fallanic I know you're busy, just pinging.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      - "25"
+    environment:
+      - SMTP_SERVER=${SMTP_SERVER:-smtp.mycompany.com}
+      - SMTP_USERNAME=${SMTP_USERNAME:-user@mycompany.com}
+      - SMTP_PASSWORD=${SMTP_PASSWORD:-changeme}
+      - SERVER_HOSTNAME=${SERVER_HOSTNAME:-localhost.localdomain}
+
+volumes:
+  redis_data:
+    #driver: local
```
Comment (fallanic) [inline docker-compose.yml:68]: shall we remove this?

```suggestion
```

**Review by fallanic — APPROVED** [review]
Looks (and works) great, thanks @tracker1 👏 🏅 

Just a question about some config commented out in `docker-compose.yml`

File: docker-compose.yml (hunk 1 above)
Comment (tracker1) [reply docker-compose.yml:68]: Can remove it in a future commit... it's mostly there for reference.

**Pull request merged**